src/main/java/com/igot/cb/access_settings/service/impl/CourseServiceImpl.java

```diff
@@ -110,9 +133,217 @@ public ApiResponse readContentState(Map<String, Object> requestBody, String auth
+            Object existingCompletedTimeObj = existingContent.get(Constants.LAST_COMPLETED_TIME);
+            Date existingCompletedTime;
+            if (existingCompletedTimeObj instanceof String) {
+                existingCompletedTime = parseDate((String) existingCompletedTimeObj);
+            } else if (existingCompletedTimeObj instanceof Date) {
+                existingCompletedTime = (Date) existingCompletedTimeObj;
+            } else {
+                existingCompletedTime = null;
+            }
+            if (existingCompletedTime == null) {
+                existingCompletedTime = parseDate((String) existingContent.getOrDefault(Constants.OLD_LAST_COMPLETED_TIME, ""));
+            }
+
+            if (inputStatus >= existingStatus) {
+                if (inputStatus >= 2) {
+                    updatedContent.put(Constants.STATUS, 2);
+                    updatedContent.put(Constants.PROGRESS, 100);
+                    updatedContent.put(Constants.LAST_COMPLETED_TIME, compareTime(existingCompletedTime, inputCompletedTime));
+                }
+            } else {
+                updatedContent.put(Constants.STATUS, existingStatus);
+            }
+        } else {
+            if (inputStatus >= 2) {
+                updatedContent.put(Constants.PROGRESS, 100);
+                updatedContent.put(Constants.LAST_COMPLETED_TIME, compareTime(null, inputCompletedTime));
+                updatedContent.put(Constants.STATUS, 2);
+            } else {
+                updatedContent.put(Constants.PROGRESS, 0);
+            }
+            updatedContent.put(Constants.LAST_ACCESS_TIME, compareTime(null, inputAccessTime));
+        }
+
+        updatedContent.put(Constants.LAST_UPDATED_TIME, Instant.now());
+        updatedContent.put(Constants.USER_ID, userId);
+        updatedContent.replaceAll((k, v) -> v instanceof Timestamp ? ((Timestamp) v).toInstant() : v);
+
+        return updatedContent;
+    }
+    public Date parseDate(String dateString) {
+        if (StringUtils.isNotBlank(dateString) && !StringUtils.equalsIgnoreCase(Constants.NULL, dateString)) {
+            try {
+                return dateFormatter.parse(dateString);
+            } catch (ParseException e) {
+                logger.error("Error parsing date: {}", dateString, e);
+                // Log and return null if date format is invalid
+                 // Replace with proper logging in production
+            }
+        }
+        return null;
+    }
+
+    // Method to map payload keys to Cassandra columns
+    public static Map<String, Object> mapPayloadToCassandraColumns(Map<String, Object> payload, Map<String, String> mapping) {
+        Map<String, Object> cassandraMap = new HashMap<>();
+        for (Map.Entry<String, Object> entry : payload.entrySet()) {
+            String cassandraKey = mapping.getOrDefault(entry.getKey(), entry.getKey());
+            cassandraMap.put(cassandraKey, entry.getValue());
+        }
+        return cassandraMap;
+    }
```

Comment:
CassandraUtil class which is reading the data from ResultSet object is already have methods to convert - use the existing one